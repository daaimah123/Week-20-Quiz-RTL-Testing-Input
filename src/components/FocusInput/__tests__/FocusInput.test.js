import React from 'react';
import FocusInput from '../FocusInput';
import { render, fireEvent, cleanup } from '@testing-library/react';

afterEach(cleanup);

test('FocusInput matches snapshot', () => {
  // set a rendered `ChangenInput` to a deconstructed `container`
  const { container } = render(<FocusInput />)
  // verify that the first instance of the component in the container matches the populated snapshot
  expect(container.firstChild).toMatchSnapshot();
});

test('clicking on button trigger focus on input', () => {
  // set a rendered `ChangenInput` to a deconstructed `getByPlaceholderText` and `getByText`
  const { getByPlaceholderText, getByText } = render(<FocusInput />)
  // simulate a click event on button text "Click to Focus"
  fireEvent.click(getByText("Click to Focus"));
  // verify that the active element within the document has the text "Focus me!"
  expect(document.activeElement).toBe(getByPlaceholderText('Focus me!'));
});