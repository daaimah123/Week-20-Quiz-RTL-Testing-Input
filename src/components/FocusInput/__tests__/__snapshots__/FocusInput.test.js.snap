// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusInput matches snapshot 1`] = `
<div>
  <input
    aria-label="focus-input"
    placeholder="Focus me!"
    type="text"
  />
  <button>
    Click to Focus
  </button>
</div>
`;
